src/tests/fs_tests.rs

```diff
@@ -2512,26 +2512,38 @@ pub mod fs_tests {
         let bufsize = 50;
         let mut vec = vec![0u8; bufsize as usize];
         let baseptr: *mut u8 = &mut vec[0];
-
+        // Remove the directory if it exists
+        let _ = cage.rmdir_syscall("/getdents");
         assert_eq!(cage.mkdir_syscall("/getdents", S_IRWXA), 0);
-        let fd = cage.open_syscall("/getdents", O_RDWR, S_IRWXA);
+        let fd = cage.open_syscall("/getdents", O_RDONLY | O_DIRECTORY, S_IRWXA);
+        assert!(fd > 0, "Failed to open directory");
+        // Check the return value of getdents_syscall
         assert_eq!(cage.getdents_syscall(fd, baseptr, bufsize as u32), 48);
+        let result = cage.getdents_syscall(fd, baseptr, bufsize as u32);
+        assert!(result >= 0, "getdents_syscall failed with error: {}", result);
 
         unsafe {
             let first_dirent = baseptr as *mut interface::ClippedDirent;
-            assert!((*first_dirent).d_off == 24);
-            let reclen_matched: bool = ((*first_dirent).d_reclen == 24);
+            // Copy packed fields into local variables to avoid alignment issues
```

Comment:
Can you explain more what this is doing?   What alignment?   Do you mean word, etc. alignment?   Why is this needed?

Reply:
This is a byte alignment issue, the fields in the packed structure (ClippedDirent) are tightly packed without the usual padding that would align them on word boundaries (like 4 or 8 bytes). Because of this, directly accessing these fields cause problems on some systems that expect memory to be accessed in larger chunks (like words).

Reply:
Can you explain this in the comment in more detail?   Once this happens, I'll approve